### PR TITLE
fix toml parsing of double type

### PR DESCRIPTION
**Pull request by bl0up**
Fix toml config file parsing for options that map to type Double. Previously setting a value in toml config for a long option type would result in, e.g. :

```
Value of 'tx-pool-limit-by-account-percentage' is a float
```

### CODE REVIEW

**Comment by macfarla** [comment]
There's one failing unit test @bl0up 
```
defaultValueForOptionMustMatchType

org.hyperledger.besu.cli.TomlConfigFileDefaultProviderTest

picocli.CommandLine$ParameterException: Invalid TOML configuration: org.apache.tuweni.toml.TomlParseError: Unexpected 'a-double-value-option', expected a newline or end-of-input (line 10, column 75)
	at app//org.hyperledger.besu.cli.util.TomlConfigFileDefaultProvider.loadConfigurationFromFile(TomlConfigFileDefaultProvider.java:158)
	at app//org.hyperledger.besu.cli.util.TomlConfigFileDefaultProvider.defaultValue(TomlConfigFileDefaultProvider.java:52)
	at app//org.hyperledger.besu.cli.TomlConfigFileDefaultProviderTest.defaultValueForOptionMustMatchType(TomlConfigFileDefaultProviderTest.java:158)
	at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.17/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.17/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

**Comment by bl0up** [comment]
I merged the latest changes from besu in my branch and now, tests that were ok before are failing. How can I fix this?

**Comment by macfarla** [comment]
> I merged the latest changes from besu in my branch and now, tests that were ok before are failing. How can I fix this?

drat, you hit a flaky acceptance test - should be ok on next run. https://app.circleci.com/pipelines/github/hyperledger/besu/19525/workflows/34c44170-8228-488a-b885-6dfa7c81f131/jobs/119210

 - so now you're just waiting on a review :)

**Comment by bl0up** [comment]
> > I merged the latest changes from besu in my branch and now, tests that were ok before are failing. How can I fix this?
> 
> drat, you hit a flaky acceptance test - should be ok on next run. https://app.circleci.com/pipelines/github/hyperledger/besu/19525/workflows/34c44170-8228-488a-b885-6dfa7c81f131/jobs/119210
> 
> * so now you're just waiting on a review :)

Yes, that was the issue. I'll wait for the review then! :)

File: besu/src/test/java/org/hyperledger/besu/cli/TomlConfigFileDefaultProviderTest.java [hunk 1]
```diff
@@ -146,6 +147,9 @@ public void defaultValueForOptionMustMatchType() throws IOException {
       fileWriter.newLine();
       fileWriter.write(
           "a-nested-multi-value-option=[ [\"value1\", \"value2\"], [\"value3\", \"value4\"] ]");
+      fileWriter.newLine();
+      fileWriter.write("a-double-value-option=0.01");
+      fileWriter.newLine();
```
Comment (mark-terry) [inline besu/src/test/java/org/hyperledger/besu/cli/TomlConfigFileDefaultProviderTest.java:152]: Is the extra newline required?

**Review by mark-terry — COMMENTED** [review]
LGTM - one query. 👍

File: besu/src/test/java/org/hyperledger/besu/cli/TomlConfigFileDefaultProviderTest.java (hunk 1 above)
Comment (bl0up) [reply besu/src/test/java/org/hyperledger/besu/cli/TomlConfigFileDefaultProviderTest.java]: Thanks! No, indeed, it's not necessary

**Review by mark-terry — APPROVED** [review]
LGTM.

**Pull request merged**